packages/common/gotrue.ts

```diff
@@ -33,10 +33,29 @@ if (debug) {
   }
 }
 
+let previousOperation: Promise<any> = Promise.resolve()
+
+export async function lockTab<R>(
+  name: string,
+  acquireTimeout: number,
+  fn: () => Promise<R>
+): Promise<R> {
+  try {
+    await previousOperation
+  } catch (e: any) {
+    // not important
+  }
+
+  previousOperation = fn
+
+  return await previousOperation
+}
+
 export const gotrueClient = new GoTrueClient({
   url: process.env.NEXT_PUBLIC_GOTRUE_URL,
   storageKey: STORAGE_KEY,
   detectSessionInUrl: true,
   debug,
   fetch: customFetch,
+  lock: globalThis.navigator && globalThis.navigator.locks ? navigatorLock : null,
```

Comment:
Consider adding a comment explaining why the globalThis.navigator.locks check is necessary. This will improve code readability. [medium]